pkg/secrets/pushtofile/secret_group.go

```diff
@@ -58,11 +59,12 @@ func (sg *SecretGroup) PushToFile(secrets []*Secret) error {
 func (sg *SecretGroup) pushToFileWithDeps(
 	depOpenWriteCloser openWriteCloserFunc,
 	depPushToWriter pushToWriterFunc,
-	secrets []*Secret) error {
+	secrets []*Secret,
+) (err error) {
```

Comment:
I'm curious about the motivation behind switched to a named return argument ?
I think we tend to avoid them because they can be a bit challenging to follow

Reply:
That's fair, they are kind of hard to follow.

I was having trouble getting the deferred function to override the returned `err` variable upon panic in `depPushToWriter`. I don't think it's capable of assigning variables that are declared in any other way ([playground example](https://play.golang.org/p/-MVgib3WVQx)).

Reply:
Thanks, TIL! Interesting behaviour, not at all the mental model I had